src/core/utils/useGetReservationGroups.ts

```diff
@@ -0,0 +1,141 @@
+import { groupBy, map, min, reduce } from "lodash";
+import { useGetReservationsV2 } from "../fbs/fbs";
+import { ReservationDetailsV2 } from "../fbs/model";
+
+/**
+ * Custom reservation details type which covers parallel reservations.
+ *
+ * Normally a reservation only covers a single record but we want to treat
+ * parallel reservations the same way as we do single reservations regarding
+ * both presentation and handling. Thus we group reservation details for
+ * individual records into a single object.
+ *
+ * We still need the individual reservations and their related records for
+ * further handling so we store these in a new map.
+ *
+ * All other values are for intent and purposes expected to the same for all
+ * reservations in the group.
+ */
+export type ReservationGroupDetails = Omit<
+  ReservationDetailsV2,
+  "reservationId" | "recordId"
+> & {
+  records: {
+    // Record id to reservation id map.
+    [key: string]: number;
+  };
+};
+
+function groupReservations(data: ReservationDetailsV2[]) {
+  const reservationGroups = groupBy(
+    data,
+    (reservation) => reservation.transactionId
+  );
+
+  const processedReserations = map(
+    reservationGroups,
+    (reservationGroup: ReservationDetailsV2[]): ReservationGroupDetails => {
+      return {
+        // Use the first reservation in the group as the base.
+        ...reservationGroup[0],
+        // The queue number for the group is the lowest individual queue number
+        numberInQueue: min(map(reservationGroup, "numberInQueue")),
+        records: reduce(
+          reservationGroup,
+          (result, reservation) => {
+            return {
+              ...result,
+              [reservation.recordId]: reservation.reservationId
+            };
+          },
+          {}
+        )
+      };
+    }
+  );
+  return processedReserations;
+}
```

Comment:
This is also another key point of the PR: Reservations are grouped by transaction id to identify and create parallel reservations.